### PR TITLE
LWG feedback from 2022-12-06

### PR DESCRIPTION
**Pull request by ericniebler**
wording review of `in_place_stop_callback`

### CODE REVIEW

File: execution.bs [hunk 1]
```diff
@@ -3498,7 +3498,7 @@ bool request_stop() noexcept;
 
 4. *Effects*: Atomically determines whether the stop state inside `*this` has received a stop request, and if not, makes a stop request.
     The determination and making of the stop request are an atomic read-modify-write operation ([intro.races]).
-    If the request was made, the callback invocations registered by associated `in_place_stop_callback` objects are executed.
+    If the request was made, the registered invocations are executed and the evaluation of the invocations are indeterminely sequenced.
```
Comment (CaseyCarter) [inline execution.bs:3501]:
```suggestion
    If the request was made, the registered invocations are executed and the evaluations of the invocations are indeterminately sequenced.
```

File: execution.bs [hunk 2]
```diff
@@ -3542,39 +3542,42 @@ Insert a new subclause, Class template `in_place_stop_callback` <b>[stopcallback
...
-3. *Effects*: Initializes `callback_` with `std::forward<C>(cb)`.
-    If `st.stop_requested()` is `true`, then `std::forward<Callback>(callback_)()` is evaluated in the current thread before the constructor returns.
-    Otherwise, if `st` has an associated `in_place_stop_source` object, registers the callback with the stop state of the `in_place_stop_source` that `st` is associated with such that `std::forward<Callback>(callback_)()` is evaluated by the first call to `request_stop()` on an associated `in_place_stop_source`.
-    The `in_place_stop_callback` object being initialized becomes associated with the `in_place_stop_source` object that `st` is associated with, if any.
+3. *Effects*: Initializes <code><i>callback_</i></code> with `std::forward<C>(cb)`.
+    Any `in_place_stop_source` associated with `st` becomes associated with `*this`.
+    The callback invocation <code>std::forward&lt;Callback>(<i>callback_</i>)()</code>
+    is *registered* ([stopsource.inplace.general]) with the associated
+    `in_place_stop_source`, if any. If the registration fails, the callback invocation
+    is evaluated in the current thread before the constructor returns.
```
Comment (ericniebler) [inline execution.bs:3564]:
```suggestion
    `in_place_stop_source`, if any. If the registration fails, evaluates
    the callback invocation.
```

File: execution.bs [hunk 3]
```diff
@@ -3542,39 +3542,42 @@ Insert a new subclause, Class template `in_place_stop_callback` <b>[stopcallback
...
 2. *Preconditions*: `Callback` and `C` model `constructible_from<Callback, C>`.
 
-3. *Effects*: Initializes `callback_` with `std::forward<C>(cb)`.
-    If `st.stop_requested()` is `true`, then `std::forward<Callback>(callback_)()` is evaluated in the current thread before the constructor returns.
-    Otherwise, if `st` has an associated `in_place_stop_source` object, registers the callback with the stop state of the `in_place_stop_source` that `st` is associated with such that `std::forward<Callback>(callback_)()` is evaluated by the first call to `request_stop()` on an associated `in_place_stop_source`.
-    The `in_place_stop_callback` object being initialized becomes associated with the `in_place_stop_source` object that `st` is associated with, if any.
+3. *Effects*: Initializes <code><i>callback_</i></code> with `std::forward<C>(cb)`.
+    Any `in_place_stop_source` associated with `st` becomes associated with `*this`.
+    The callback invocation <code>std::forward&lt;Callback>(<i>callback_</i>)()</code>
+    is *registered* ([stopsource.inplace.general]) with the associated
```
Comment (ericniebler) [inline execution.bs:3562]:
```suggestion
    Registers ([stopsource.inplace.general]) the callback invocation
    <code>std::forward&lt;Callback>(<i>callback_</i>)()</code> with the associated
```

File: execution.bs [hunk 4]
```diff
@@ -3542,39 +3542,42 @@ Insert a new subclause, Class template `in_place_stop_callback` <b>[stopcallback
...
+    `in_place_stop_source`, if any. If the registration fails, the callback invocation
+    is evaluated in the current thread before the constructor returns.
 
-4. *Throws*: Any exception thrown by the initialization of `callback_`.
+4. *Throws*: Any exception thrown by the initialization of <code><i>callback_</i></code>.
 
-5. *Remarks*: If evaluating `std::forward<Callback>(callback_)()` exits via an exception, then `terminate` is invoked ([except.terminate]).
+5. *Remarks*: If evaluating <code>std::forward&lt;Callback>(<i>callback_</i>)()</code>
+    exits via an exception, then `terminate` is invoked ([except.terminate]).
```
Comment (ericniebler) [inline execution.bs:3569]: Add to the list in [except.terminate]

File: execution.bs (hunk 1 above)
Comment (ericniebler) [inline execution.bs:3501]:
```suggestion
    If the request was made, the registered invocations are executed and the evaluations of the invocations are indeterminely sequenced.
```

File: execution.bs [hunk 5]
```diff
@@ -3542,39 +3542,42 @@ Insert a new subclause, Class template `in_place_stop_callback` <b>[stopcallback
...
 
 <pre highlight=c++>
 ~in_place_stop_callback();
 </pre>
 
-6. *Effects*: Unregisters the callback from the stop state of the associated `in_place_stop_source` object, if any.
-    The destructor does not block waiting for the execution of another callback registered by an associated `stop_callback`.
-    If `callback_` is concurrently executing on another thread, then the return from the invocation of `callback_` strongly happens before ([intro.races]) `callback_` is destroyed.
-    If `callback_` is executing on the current thread, then the destructor does not block ([defns.block]) waiting for the return from the invocation of `callback_`.
+6. *Effects*: *Unregisters* ([stopsource.inplace.general]) the callback invocation from
```
Comment (ericniebler) [inline execution.bs:3575]:
```suggestion
6. *Effects*: Unregisters ([stopsource.inplace.general]) the callback invocation from
```

**Pull request merged**